lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -113,5 +113,18 @@ public void OnCompleted()
         public void OnError(Exception error)
         {
         }
+
+        /// <summary>
+        /// Observer of IDriverMetrics.
+        /// When Driver metrics data is changed, this method will be called
+        /// </summary>
+        /// <param name="value"></param>
+        public void OnNext(IDriverMetrics value)
+        {
+            var set = new HashSet<KeyValuePair<string, string>>();
```

Comment:
I still don't understand why it should be a `Set` and not, say, an array. Do we have a strict requirement that all `KeyValuePair` values should be unique?